Svc/DpCatalog/DpCatalog.cpp

```diff
@@ -618,27 +702,144 @@ bool DpCatalog::allocateNode(DpBtreeNode*& newNode, const DpStateEntry& newEntry
         this->log_WARNING_HI_DpCatalogFull(newEntry.record);
         return false;
     }
+
     // get a new node from the free list
     newNode = this->m_freeListHead;
     // move the head of the free list to the next node
+    // If we've at the bottom of the free list, head will now be nullptr
     this->m_freeListHead = this->m_freeListHead->left;
 
     // initialize the new node
     newNode->left = nullptr;
     newNode->right = nullptr;
+    newNode->parent = nullptr;
     newNode->entry = newEntry;
 
     // we got one, so return success
     return true;
 }
 
-void DpCatalog::deleteEntry(DpStateEntry& entry) {}
+void DpCatalog::deallocateNode(DpBtreeNode* node) {
+    DpBtreeNode* parent = node->parent;
+
+    // since nodes are deallocated after xmit, left should be gone
+    // However, left node could be added during xmit
+    if (node->left != nullptr) {
+        // To try to only deallocate nodes at fileDone,
+        // finished nodes w/ children are trimmed out
+        // Simple shifts only work with branches that don't branch
+
+        // Since we aren't limited to adding just 1 node during file transfer
+        // left might not be a leaf
+        // Instead, find the lowest priority node on the left branch
```

Comment:
```suggestion
        // Instead, find the highest priority node on the left branch
```
I think we want the rightmost node under the left child. I'm assuming priorities go up from left to right.

Reply:
It's a bit convoluted. I believe my original comment is actually correct.
Here is my understanding of the priority system:

For all compared fields (i.e. priority, tSec, tSub, id), a lower numerical value is treated as higher priority. To keep with this scheme, my compareEntries function returns a -1 if the first entry passed in is less than the second entry (meaning it is numerically lower and thus higher priority). My comparison operators return numerical rather than priority comparisons (e.g. an entry filled with all zeros is "less than" a node filled with all ones even though it is higher priority). The insertion scheme places higher priority nodes (numerically lesser) on the left branch.

This step is looking to replace the just transmitted (and shortly deallocated) node w/ a node of immediately neighboring priority. Since the left (higher priority) side can only be added to by the addToCat port, there should be fewer nodes to traverse (than the lower priority half of the catalog) making the lowest priority child of the left node the best candidate.